rules/framework.bzl

```diff
@@ -274,13 +272,42 @@ def _apple_framework_packaging_impl(ctx):
         )
         _add_to_dict_if_present(objc_provider_fields, key, set)
 
-    objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
-    swift_provider = swift_common.create_swift_info(swift_infos=swift_infos)
-    default_info_provider = DefaultInfo(files = depset(framework_files))
-    # vfs_overlay_provider = VFSOverlay(
-    #     files = depset(items = file_map, transitive = [dep[VFSOverlay].files for dep in ctx.attr.transitive_deps if VFSOverlay in dep])
-    # )
-    return [objc_provider, swift_provider, default_info_provider]
+    # gather swift info fields
+    swift_info_fields = {
+        "swift_infos": [dep[SwiftInfo] for dep in ctx.attr.transitive_deps if SwiftInfo in dep],
```

Comment:
for each of the transitive dependencies of ctr.attr (the current context we are in), if there is SwiftInfo in dep, take it out and add to an array. This array then becomes the value with key = "swift_infos"